src/main/java/duke/TodoList.java

```diff
@@ -0,0 +1,67 @@
+package duke;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+
+public class TodoList implements Serializable {
+    private ArrayList<Task> todo_list;
+
+    public TodoList() {
+        this.todo_list = new ArrayList<>(100);
+    }
+
+    public void add(String type, String task) throws DukeExceptions{
+        Task taskObject = new Task();
+        Task newTask = taskObject.createNewTask(type, task);
+        System.out.println("Got it. I've added this task:");
+        todo_list.add(newTask);
+        System.out.println(newTask);
+    }
+
+    public void mark(int index) throws DukeExceptions{
+        int todo_list_length = todo_list.size();
+        if (index < 0 || index > todo_list_length) {
+            throw new DukeExceptions("Please use list command to check the index!");
+        }
+        Task task = todo_list.get(index - 1);
+        task.markTask();
+        System.out.println("Nice! I've marked this task as done:");
+        System.out.println(todo_list.get(index-1));
```

Comment:
Perhaps you can move `System.out.println(..)` into a function in your Ui class to utilise it more, since the purpose of the Ui class is to handle all interactions with the user